flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/TaskMailboxImpl.java

```diff
@@ -185,31 +171,19 @@ private Mail takeHeadInternal(int priority) throws IllegalStateException {
 		}
 	}
 
-	private boolean isEmpty() {
-		return count == 0;
-	}
-
-	private boolean isPutAbleState() {
-		return state == State.OPEN;
-	}
-
-	private boolean isTakeAbleState() {
-		return state != State.CLOSED;
-	}
-
 	private void checkPutStateConditions() {
 		final State state = this.state;
-		if (!isPutAbleState()) {
+		if (this.state != OPEN) {
```

Comment:
I mean instead `if (condition) throw new IllegalStateException(msg)` you can just replace it with `checkState(condition, msg)?

But ok, `checkPutStateConditions` could be on the hot path, so it might be a good idea to hide creating the `msg` inside the `if` branch, so the message is not created every time.

Reply:
Since it's on the hot path, I'd leave it as is.